lib/gooddata/lcm/actions/synchronize_users.rb

```diff
@@ -252,17 +252,17 @@ def call(params)
                                            do_not_touch_users_that_are_not_mentioned: do_not_touch_users_that_are_not_mentioned,
                                            create_non_existing_user_groups: create_non_existing_user_groups)
                     when 'sync_multiple_projects_based_on_custom_id'
-                      new_users.group_by { |u| u[:pid] }.flat_map do |client_id, users|
+                      new_users.group_by { |u| u[:pid] }.flat_map.pmap do |client_id, users|
                         fail "Client id cannot be empty" if client_id.blank?
                         begin
-                          project = domain.clients(client_id, data_product).project
+                          current_project = domain.clients(client_id, data_product).project
                         rescue RestClient::BadRequest => e
                           raise e unless /does not exist in data product/ =~ e.response
                           fail "The client \"#{client_id}\" does not exist in data product \"#{data_product.data_product_id}\""
                         end
-                        fail "Client #{client_id} does not have project." unless project
-                        puts "Project #{project.pid} of client #{client_id} will receive #{users.count} users"
-                        project.import_users(users,
+                        fail "Client #{client_id} does not have project." unless current_project
+                        puts "Project #{current_project.pid} of client #{client_id} will receive #{users.count} users"
+                        current_project.import_users(users,
```

Comment:
warning: Similar code found in synchronize_users.rb:193, synchronize_users.rb:204, synchronize_users.rb:221, synchronize_users.rb:247, synchronize_users.rb:265, synchronize_users.rb:292, synchronize_users.rb:334 (mass = 112) (flay)